src/backend/Level0/CHIPBackendLevel0.cc

```diff
@@ -636,99 +670,100 @@ void CHIPCallbackEventMonitorLevel0::monitor() {
-     * EventCommandListMap
-     */
-    if (Stop) {
-      // get current host time in seconds
-      int CurrTime = std::chrono::duration_cast<std::chrono::seconds>(
-                         std::chrono::system_clock::now().time_since_epoch())
-                         .count();
-
-      // if this is the first time we are stopping, set the time
-      if (TimeSinceStopRequested_ == 0)
-        TimeSinceStopRequested_ = CurrTime;
-
-      int EpasedTime = CurrTime - this->TimeSinceStopRequested_;
-      bool AllEventsCleared = Backend->Events.size() == 0;
-      if (AllEventsCleared) {
-        pthread_exit(0);
-      } else if (EpasedTime > BackendZe->getCollectEventsTimeout()) {
-        logError(
-            "CHIPStaleEventMonitorLevel0 stop was called but not all events "
-            "have been cleared. Timeout of {} seconds has been reached.",
-            BackendZe->getCollectEventsTimeout());
-        int MaxPrintEntries = 10;
-        int PrintedEntries = 0;
-        for (auto &Event : Backend->Events) {
-          auto EventLz = std::static_pointer_cast<CHIPEventLevel0>(Event);
-          logError("Uncollected Backend->Events: {} {} AssocCmdList {}",
-                   (void *)Event.get(), Event->Msg,
-                   (void *)EventLz->getAssocCmdList());
-          PrintedEntries++;
-          if (PrintedEntries > MaxPrintEntries)
-            break;
-        }
+  } // done collecting events to delete
+}
 
-        pthread_exit(0);
-      } else {
-        // print only once a second to avoid saturating stdout with logs
-        if (CurrTime - LastPrint_ >= 1) {
-          LastPrint_ = CurrTime;
-          logDebug("CHIPStaleEventMonitorLevel0 stop was called but not all "
-                   "events have been cleared. Timeout of {} seconds has not "
-                   "been reached yet. Elapsed time: {} seconds",
-                   BackendZe->getCollectEventsTimeout(), EpasedTime);
-        }
+void CHIPStaleEventMonitorLevel0::exitChecks_() {
+  LOCK(EventMonitorMtx); // chipstar::EventMonitor::Stop
+  CHIPBackendLevel0 *BackendZe = static_cast<CHIPBackendLevel0 *>(Backend);
+  /**
+   * In the case that a user doesn't destroy all the
+   * created streams, we remove the streams and outstanding events in
+   * Backend::waitForThreadExit() but Backend has no knowledge of
+   * EventCommandListMap
+   */
+  if (Stop) {
+    // get current host time in seconds
+    int CurrTime = std::chrono::duration_cast<std::chrono::seconds>(
+                       std::chrono::system_clock::now().time_since_epoch())
+                       .count();
+
+    // if this is the first time we are stopping, set the time
+    if (TimeSinceStopRequested_ == 0)
+      TimeSinceStopRequested_ = CurrTime;
+
+    int EpasedTime = CurrTime - this->TimeSinceStopRequested_;
+    bool AllEventsCleared = Backend->Events.size() == 0;
+    if (AllEventsCleared) {
+      pthread_exit(0);
+    } else if (EpasedTime > BackendZe->getCollectEventsTimeout()) {
```

Comment:
[Else-after-return](https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return). 
```suggestion
    if (AllEventsCleared)
      pthread_exit(0);

    if (EpasedTime > BackendZe->getCollectEventsTimeout()) {
```
There is another one in the below too.